src/Streamly/Internal/Unicode/Utf8/Reduce.hs

```diff
@@ -0,0 +1,334 @@
+    , chunksOf
+
+    -- * Searching
+    , breakOnAll
+    , partition
+    )
+where
+
+#include "inline.hs"
+
+--------------------------------------------------------------------------------
+-- Imports
+--------------------------------------------------------------------------------
+
+import System.IO.Unsafe (unsafePerformIO)
+
+import qualified Streamly.Internal.Data.Stream.IsStream as Stream
+
+import Streamly.Internal.Unicode.Utf8.Type
+import Streamly.Internal.Unicode.Utf8.Transform (filter)
+
+import Prelude hiding
+    ( all
+    , any
+    , break
+    , concat
+    , concatMap
+    , drop
+    , dropWhile
+    , elem
+    , filter
+    , foldl
+    , foldl1
+    , foldr
+    , foldr1
+    , head
+    , init
+    , last
+    , length
+    , lines
+    , map
+    , maximum
+    , minimum
+    , null
+    , read
+    , replicate
+    , reverse
+    , scanl
+    , scanl1
+    , scanr
+    , scanr1
+    , span
+    , splitAt
+    , tail
+    , take
+    , takeWhile
+    , unlines
+    , unwords
+    , words
+    , zip
+    , zipWith
+    )
+
+-- $setup
+-- >>> :set -XOverloadedStrings
+-- >>> import qualified Streamly.Internal.Unicode.Utf8 as Utf8
+
+--------------------------------------------------------------------------------
+-- Substrings
+--------------------------------------------------------------------------------
+
+-- | 'splitAt' @n t@ returns a pair whose first element is a
+-- prefix of @t@ of length @n@, and whose second is the remainder of
+-- the string. It is equivalent to @('take' n t, 'drop' n t)@.
+--
+-- /Time complexity:/ O(n)
+splitAt :: Int -> Utf8 -> (Utf8, Utf8)
+splitAt = undefined
+
+-- | 'span', applied to a predicate @p@ and text @t@, returns
+-- a pair whose first element is the longest prefix (possibly empty)
+-- of @t@ of elements that satisfy @p@, and whose second is the
+-- remainder of the list.
+--
+-- >> Utf8.span (=='0') "000AB"
+-- ("000","AB")
+--
+-- /Time complexity:/ O(n)
+{-# INLINE span #-}
+span :: (Char -> Bool) -> Utf8 -> (Utf8, Utf8)
+span = undefined
+
+-- | 'break' is like 'span', but the prefix returned is
+-- over elements that fail the predicate @p@.
+--
+-- >> Utf8.break (=='c') "180cm"
+-- ("180","cm")
+--
+-- /Time complexity:/ O(n)
+{-# INLINE break #-}
+break :: (Char -> Bool) -> Utf8 -> (Utf8, Utf8)
+break p = span (not . p)
+
+-- | Group characters in a string according to a predicate.
+--
+-- /Time complexity:/ O(n)
+groupBy :: (Char -> Char -> Bool) -> Utf8 -> [Utf8]
+groupBy p = unsafePerformIO . Stream.toList . Stream.groupsBy p write . toStream
+
+{-
+-- | Returns the /array/ index (in units of 'Word16') at which a
+-- character may be found.  This is /not/ the same as the logical
+-- index returned by e.g. 'findIndex'.
+findAIndexOrEnd :: (Char -> Bool) -> Utf8 -> Int
+findAIndexOrEnd q t@(Utf8 _arr _off len) = go 0
+    where go !i | i >= len || q c       = i
+                | otherwise             = go (i+d)
+                where Iter c d          = iter t i
+-}
```

Comment:
We can remove this, or just keep the signature if you really want to add it later.